File: ControlGastos/app/src/main/java/com/example/controlgastos/ListCategory.java
package com.example.controlgastos;

import java.util.LinkedList;
import java.util.List;

public class ListCategory extends ListNode {

    private List<ListNode> nodes;

    public ListCategory(String name, List<ListNode> nodes) {
        super(name);
        this.nodes = nodes;
    }

    public List<ListNode> getNodes() {
        return nodes;
    }

    public void setNodes(List<ListNode> nodes) {
        this.nodes = nodes;
    }

    @Override
    double getPrecio() {
        double precio = 0;
        for (ListNode node : nodes) {
            precio += node.getPrecio();
        }
        return precio;
    }
}

File: ControlGastos/app/src/main/java/com/example/controlgastos/ListCategory.kt
package com.example.controlgastos

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class ListCategory(override val name: String, val nodes: MutableList<ListNode>) :
    ListNode(name, ListNodeType.CATEGORY) {

    constructor() : this("", mutableListOf<ListNode>())

    constructor(name: String) : this(name, mutableListOf<ListNode>())

    @PrimaryKey
    var id: Int = 0

    override fun getTotal(): Double {
        return nodes.sumOf { it.getTotal() }
    }

    fun addChildren(node: ListNode) {
        nodes.add(node)
    }

}